projects/frontend/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -10,32 +15,163 @@ import { environment } from 'projects/frontend/src/environments/environment';
 export class DashboardComponent implements OnInit {
 
   private apiUrl: string;
-  constructor(private http: HttpClient) {
+
+  constructor(
+    private http: HttpClient,
+    private authService: AuthService,
+    private taskService: TaskService,
+    private formBuilder: FormBuilder
+    ) {
     this.apiUrl = environment.API_URL;
+    // if (this.authService.getLoggedInUser() == null) {
+    //   this.router.navigate(['signin']);
+    // }
   }
 
   tasks;
-  all;
-  completed;
-  ongoing;
-  upcoming;
-  percentComp;
+  currentUser;
+  addform: FormGroup;
+  updateform: FormGroup;
+  loading = false;
+  updating = false;
+  adding = false;
+  task: Task;
+  etask: Task;
 
   ngOnInit(): void {
+    this.getUser();
+    this.addform = this.formBuilder.group({
+      task: ['', [Validators.required]],
+      priority: ['', [Validators.required]],
+      status: ['', [Validators.required]]
+    });
+    this.updateform = this.formBuilder.group({
+      task: ['', [Validators.required]],
+      priority: ['', [Validators.required]],
+      status: ['', [Validators.required]]
+    });
+  }
+
+  get af() { return this.addform.controls; }
+  get uf() { return this.updateform.controls; }
+
+  getUser() {
+    this.currentUser = this.authService.getLoggedInUser();
     this.listTasks();
   }
+  viewTask(id) {
+    this.taskService.getTask(id).subscribe(
```

Comment:
Use Angular Template Binding

Reply:
Changed to use angular template binding.